src/patch.ml

```diff
@@ -247,18 +247,29 @@ let pp ~git ppf {operation; hunks; mine_no_nl; their_no_nl} =
 let pp_list ~git ppf diffs =
   List.iter (Format.fprintf ppf "%a" (pp ~git)) diffs
 
+let process_git_prefix ~git ~prefix s =
+  if git && String.is_prefix ~prefix s then
+    String.slice ~start:(String.length prefix) s
+  else
+    s
+
```

Comment:
The prefix should never be taken out without input from the user. See https://github.com/hannesm/patch/pull/9
```suggestion
```